services/consensuscontext/test/create_block_test.go

```diff
@@ -36,8 +36,8 @@ func TestRetryWhenNotEnoughTransactionsPendingOnTransactionPool(t *testing.T) {
 
 		txCount := h.config.ConsensusContextMinimumTransactionsInBlock() - 1
 
-		h.expectTransactionsRequestedFromTransactionPool(0)
-		h.expectTransactionsRequestedFromTransactionPool(txCount)
+		h.expectTxPoolToReturnXTransactions(0)
```

Comment:
why two expectations here?

Reply:
The test invokes a mocked `GetTransactionsForOrdering` that on its first call returns no tx and on its second call returns some tx (this is what the 2 expects() do), thus testing the functionality of retrying when no tx returned the first time.